src/ManageCourses.Api/Mapping/CourseMapper.cs

```diff
@@ -16,6 +16,13 @@ public class CourseMapper : ICourseMapper
 
         public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Domain.Models.Provider ucasProviderData, Domain.Models.Course ucasCourseData, ProviderEnrichmentModel providerEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel)
         {
+            // if none of the course-sites are published, return null to prevent the whole course from being published to find
+            if (ucasCourseData?.CourseSites == null || ucasCourseData.CourseSites.All(cs =>
+                    !cs.Publish.Equals("Y", StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
all courses sites need to be published?

Reply:
read as "if they are all not published"
I couldn't work out how to make this read nicely in under 5 mins, I'll give it some more thought

maybe extracting a method `HasAnySitesPublished(ucasCourseData)` would help here

Reply:
https://github.com/DFE-Digital/manage-courses-backend/pull/432

i think it needs to be tackled here over this end

Reply:
I've made the code much clearer, take another look